viewing_party/party.py

```diff
@@ -0,0 +1,114 @@
+def create_movie(movie_title, genre, rating):
+   return {"title":movie_title, "genre":genre, "rating":rating} if (movie_title and genre and rating) else None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data,title):
+    wl = user_data["watchlist"]
+    for item in wl:
+        if item["title"] == title:
+            add_to_watched(user_data,wl.pop(wl.index(item)))
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    cumulative_ratings = 0
+    watched = user_data["watched"]
+    for item in watched:
+        cumulative_ratings += item["rating"]
+
+    return cumulative_ratings/len(watched) if len(watched) > 0 else 0
+
+def get_most_watched_genre(user_data):
+    counts = {}
+    watched = user_data["watched"]
+    popular = ['',0]
+    for item in watched:
+        this_genre_count = item["genre"]
+        if this_genre_count not in counts:
+            counts[this_genre_count] = 1
+        else:
+            counts[this_genre_count] += 1
+        if counts[this_genre_count] > popular[1]:
+            popular[0] = this_genre_count
+            popular[1] = counts[this_genre_count]
+
+    return popular[0] if popular[1] > 0 else None
+
+def get_unique_watched(user_data):
+    watched = user_data["watched"]
+    uniques = watched.copy()
+    for friend in user_data["friends"]:
+        for item in friend["watched"]:
+            if item in watched and item in uniques:
```

Comment:
Something to be careful of is that the `in` and `not in` operations on a list must iterate over the entire list to check for the item (O(n)). So here we end up with a triply-nested loop. We can't really avoid the two outer loops, since we need to look at each movie of each friend, but could we use a different data structure, maybe combined with another approach, to build up some additional data to help us avoid needing the implicit innermost loop?